File: frontend/src/socket_server/SocketServer.ts
import { io } from "socket.io-client";

const socket = io();

socket.on("connect", () => {
    console.log("🔥 CONNECTED:", socket.id);
});

socket.on("disconnect", (reason) => {
    console.log("🔥 DISCONNECTED:", socket.id, reason);
});

socket.on("connect_error", (error) => {
    console.log("🔥 SOCKET ERROR:", error.message);
});

socket.onAny((event, ...args) => {
    console.log("🔥 EVENT:", event, args);
});

export default socket;

File: services/disaster-service/socket_connection/socket_server.py
import socketio

sio = socketio.AsyncServer(
    async_mode="asgi",
    cors_allowed_origins="*"
)


@sio.event
async def connect(sid, environ, auth):
    print("🔥 SOCKET CONNECTED:", sid)


@sio.event
async def disconnect(sid):
    print("🔥 SOCKET DISCONNECTED:", sid)